utils/s2n_mem.c

```diff
@@ -48,10 +48,10 @@ static int s2n_mem_init_impl(void)
     long sysconf_rc = sysconf(_SC_PAGESIZE);
 
     /* sysconf must not error, and page_size cannot be 0 */
-    POSIX_ENSURE(sysconf_rc > 0, S2N_FAILURE);
+    POSIX_ENSURE_GT(sysconf_rc, 0);
 
     /* page_size must be a valid uint32 */
-    POSIX_ENSURE(sysconf_rc <= UINT32_MAX, S2N_FAILURE);
+    POSIX_ENSURE_LTE(sysconf_rc, UINT32_MAX);
```

Comment:
Do you know why `POSIX_ENSURE` lets you specify the error code, but `POSIX_ENSURE_LTE`, etc always uses `S2N_ERR_SAFETY`?

Reply:
POSIX_ENSURE_LTE(x, y) is just a shorthand for POSIX_ENSURE(x <= y, S2N_ERR_SAFETY). If you want to specify an error other than S2N_ERR_SAFETY, you use POSIX_ENSURE.